mmdet/datasets/crowdhuman.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import json
+import logging
+import os
+import os.path as osp
+from typing import List, Union
+
+import mmcv
+from mmengine.dataset import BaseDataset
+from mmengine.fileio import FileClient
+from mmengine.logging import print_log
+
+from mmdet.registry import DATASETS
+
+
+@DATASETS.register_module()
+class CrowdHumanDataset(BaseDataset):
+    r"""Dataset for CrowdHuman.
+
+    Args:
+        data_root (str): The root directory for
+        ``data_prefix`` and ``ann_file``.
+        ann_file (str): Annotation file path.
+    """
+
+    METAINFO = {
+        'CLASSES': ('person', ),
+        # PALETTE is a list of color tuples, which is used for visualization.
+        'PALETTE': [(220, 20, 60)]
+    }
+
+    def __init__(self, data_root, ann_file, **kwargs):
+        # id_hw file is an additional annotation information which record the
+        # size of each image. This file is automatically created when you
+        # first load the CrowdHuman dataset by mmdet
+        ann_file_name = os.path.basename(ann_file)
+        if 'train' in ann_file_name:
```

Comment:
It's not a good idea to judge the dataset type like this.
For example, when the annfile path is "train_crowdhuman/annotation_val.odgt", the dataset type will always be 'train'.